### PR TITLE
KOTOR: Fix crash on head change in character generator

### PR DESCRIPTION
**Pull request by seedhartha**

### CODE REVIEW

File: src/engines/kotor/gui/chargen/chargeninfo.cpp [hunk 1]
```diff
@@ -201,7 +201,10 @@ void CharacterGenerationInfo::recreateHead() {
 	Graphics::Aurora::Model *head = _head;
 	_head = loadModelObject(Creature::getHeadMeshString(getGender(), getSkin(), getFace()), "");
 	_body->attachModel("headhook", _head);
+
+	GfxMan.lockFrame();
 	delete head;
+	GfxMan.unlockFrame();
```
Comment (DrMcCoy) [inline src/engines/kotor/gui/chargen/chargeninfo.cpp]: The attachModel() call should probably be moved within the locking as well

Comment (clone2727) [reply src/engines/kotor/gui/chargen/chargeninfo.cpp]: We should probably have an RAII class for handling the lock/unlock (if we don't already).

**Comment by seedhartha** [comment]
Done.

**Comment by DrMcCoy** [comment]
Merged as 71c97acf8, thanks! :)

**Pull request closed without merging**